### DISCUSSION TITLE
NextJS "unstable_cache" error while attempting to JSON.parse "undefined" value

### DISCUSSION THREAD

**Discussion by prasad-pul** [vercel/next.js#59087] [Help]
### Summary

I'm using the `unstable_cache` function in my application to cache the response from an expensive api call.

The application seems to be crashing with the error: **Unexpected token u in JSON at position 0**.

Little bit of investigation showed that this error is occurring inside the unstable_cache function at this point 👇
<img width="1476" alt="Screenshot 2023-11-29 at 3 36 17 PM" src="https://github.com/vercel/next.js/assets/84544020/1fe7d145-b713-45d7-a357-9a33d767101d">

Any chance that this could be because the `resData.body` could potentially be `undefined` and it is NOT being checked before `JSON.parse` is invoked on it?

**Comment by mattddean** [comment]
I think the fix is pretty simple https://github.com/vercel/next.js/pull/59126
